### PR TITLE
feat(bigquery): augment retry predicate to support additional errors

### PR DESCRIPTION
**Pull request by shollyman**
Based on similar refactor in storage:
https://github.com/googleapis/google-cloud-go/pull/4019

Fixes: https://github.com/googleapis/google-cloud-go/issues/4021

### CODE REVIEW

File: bigquery/bigquery.go [hunk 1]
```diff
@@ -177,13 +179,32 @@ func retryableError(err error) bool {
...
+	case interface{ Temporary() bool }:
+		if e.Temporary() {
+			return true
+		}
 	}
-	var reason string
-	if len(e.Errors) > 0 {
-		reason = e.Errors[0].Reason
+	// Unwrap is only supported in go1.13.x+
+	if e, ok := err.(interface{ Unwrap() error }); ok {
```
Comment (codyoss) [inline bigquery/bigquery.go:206]: nit: could this just be another case and false be the default?

**Review by codyoss — APPROVED** [review]
LGTM, just on small comment.

File: bigquery/bigquery.go (hunk 1 above)
Comment (tritone) [reply bigquery/bigquery.go:206]: Unfortunately this doesn't work-- *url.Error can also wrap errors that end up being retryable. So, we need to execute both cases, which doesn't work in a switch.

**Review by tritone — COMMENTED** [review]
Looks good! We also added io.ErrUnexpectedEOF for storage; not sure whether or not that makes sense for you.

**Review by tritone — APPROVED** [review]

File: bigquery/bigquery.go (hunk 1 above)
Comment (shollyman) [reply bigquery/bigquery.go:206]: The difference is subtle; if we have a wrapped error that matched one of the previous cases but didn't qualify for retry, it allows another chance to unwrap and re-evaluate the inner error.

**Comment by shollyman** [comment]
> Looks good! We also added io.ErrUnexpectedEOF for storage; not sure whether or not that makes sense for you.

Haven't seen this in the wild, but I see there's a few common imports that traffic it (http and grpc).  I'll add it.

**Pull request merged**